From b3ba4e9e3d963da793d7494206c9cb7fa03f8480 Mon Sep 17 00:00:00 2001
From: Robert Mader <robert.mader@collabora.com>
Date: Tue, 16 Jun 2026 15:27:34 +0200
Subject: [PATCH] gl-shaders: Allow SHADER_VARIANT_RGBA for YUV->RGB
 conversion

This assert is hit on drivers supporting YUV formats as GL_TEXTURE_2D
texture target instead of GL_TEXTURE_EXTERNAL_OES. While uncommon, it is
completely valid and we explicitly support it in related code paths.

Fixes: 82d629741 (gl-shaders: Remove asserts relying on shader compiler behavior)
Signed-off-by: Robert Mader <robert.mader@collabora.com>

Upstream-Status: Backport [https://gitlab.freedesktop.org/wayland/weston/-/commit/b3ba4e9e3d963da793d7494206c9cb7fa03f8480]
---
 libweston/renderer-gl/gl-shaders.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libweston/renderer-gl/gl-shaders.c b/libweston/renderer-gl/gl-shaders.c
index 045c09b4..d4f32550 100644
--- a/libweston/renderer-gl/gl-shaders.c
+++ b/libweston/renderer-gl/gl-shaders.c
@@ -824,7 +824,8 @@ gl_shader_load_config_representation(struct weston_compositor *compositor,
 
 	if (shader->yuv_coefficients_uniform == -1 ||
 	    shader->yuv_offsets_uniform == -1) {
-		assert(shader->key.variant == SHADER_VARIANT_EXTERNAL);
+		assert(shader->key.variant == SHADER_VARIANT_EXTERNAL ||
+		       shader->key.variant == SHADER_VARIANT_RGBA);
 		return;
 	}
 
-- 
2.54.0
